File: dataset/CoT - Copy/CLAUDE_claude-sonnet-4-5-20250929/Task10.cpp

#include <iostream>
#include <algorithm>

class Task10 {
public:
    static int getSum(int a, int b) {
        // Handle case where a and b are equal
        if (a == b) {
            return a;
        }
        
        // Ensure min and max are correctly ordered
        int minVal = std::min(a, b);
        int maxVal = std::max(a, b);
        
        // Use arithmetic series formula: sum = n * (first + last) / 2
        // where n is the count of numbers
        int count = maxVal - minVal + 1;
        return count * (minVal + maxVal) / 2;
    }
};

int main() {
    // Test cases
    std::cout << Task10::getSum(1, 0) << std::endl;    // Expected: 1
    std::cout << Task10::getSum(1, 2) << std::endl;    // Expected: 3
    std::cout << Task10::getSum(0, 1) << std::endl;    // Expected: 1
    std::cout << Task10::getSum(1, 1) << std::endl;    // Expected: 1
    std::cout << Task10::getSum(-1, 0) << std::endl;   // Expected: -1
    std::cout << Task10::getSum(-1, 2) << std::endl;   // Expected: 2
    
    return 0;
}

File: dataset/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Task10.java

public class Task10 {
    public static int getSum(int a, int b) {
        if (a == b) {
            return a;
        }
        
        int min = Math.min(a, b);
        int max = Math.max(a, b);
        
        // Sum formula: n * (first + last) / 2
        int count = max - min + 1;
        return count * (min + max) / 2;
    }
    
    public static void main(String[] args) {
        System.out.println(getSum(1, 0));    // Expected: 1
        System.out.println(getSum(1, 2));    // Expected: 3
        System.out.println(getSum(0, 1));    // Expected: 1
        System.out.println(getSum(1, 1));    // Expected: 1
        System.out.println(getSum(-1, 0));   // Expected: -1
        System.out.println(getSum(-1, 2));   // Expected: 2
    }
}
